serde/src/de/impls.rs

```diff
@@ -916,17 +916,19 @@ impl<T, E> Deserialize for Result<T, E> where T: Deserialize, E: Deserialize {
             {
                 match try!(visitor.visit_variant()) {
                     Field::Ok => {
-                        let (value,) = try!(visitor.visit_seq(TupleVisitor1::new()));
+                        let (value,) = try!(visitor.visit_tuple(1, TupleVisitor1::new()));
```

Comment:
uhm... I see the serialization change, but the deserialization here is still the same.